Fake CHT: Error: EACCES: permission denied, mkdir

I get an error when I [follow the steps](https://github.com/medic/cht-monitoring/tree/8114_metrics_mvp/tests) to run the Fake CHT:

```
fake-cht_1       | npm ERR! code EACCES
fake-cht_1       | npm ERR! syscall mkdir
fake-cht_1       | npm ERR! path /home/node/app/node_modules
fake-cht_1       | npm ERR! errno -13
fake-cht_1       | npm ERR! Error: EACCES: permission denied, mkdir '/home/node/app/node_modules'
fake-cht_1       | npm ERR!  [Error: EACCES: permission denied, mkdir '/home/node/app/node_modules'] {
fake-cht_1       | npm ERR!   errno: -13,
fake-cht_1       | npm ERR!   code: 'EACCES',
fake-cht_1       | npm ERR!   syscall: 'mkdir',
fake-cht_1       | npm ERR!   path: '/home/node/app/node_modules'
fake-cht_1       | npm ERR! }
fake-cht_1       | npm ERR! 
fake-cht_1       | npm ERR! The operation was rejected by your operating system.
fake-cht_1       | npm ERR! It is likely you do not have the permissions to access this file as the current user
fake-cht_1       | npm ERR! 
fake-cht_1       | npm ERR! If you believe this might be a permissions issue, please double-check the
fake-cht_1       | npm ERR! permissions of the file and its containing directories, or try running
fake-cht_1       | npm ERR! the command again as root/Administrator.
fake-cht_1       | 
fake-cht_1       | npm ERR! A complete log of this run can be found in: /home/node/.npm/_logs/2023-04-13T05_48_40_918Z-debug-0.log
```

@mrjones-plip I got a very similar error when running the latest from `main`:

```
npm ERR! code EACCES
2023-04-21T18:50:06.749013326Z npm ERR! syscall open
2023-04-21T18:50:06.749165537Z npm ERR! path /home/node/app/package-lock.json
2023-04-21T18:50:06.749485831Z npm ERR! errno -13
2023-04-21T18:50:06.751328210Z npm ERR! Error: EACCES: permission denied, open '/home/node/app/package-lock.json'
2023-04-21T18:50:06.751338933Z npm ERR!  [Error: EACCES: permission denied, open '/home/node/app/package-lock.json'] {
2023-04-21T18:50:06.751414290Z npm ERR!   errno: -13,
2023-04-21T18:50:06.751431295Z npm ERR!   code: 'EACCES',
2023-04-21T18:50:06.751445770Z npm ERR!   syscall: 'open',
2023-04-21T18:50:06.751465183Z npm ERR!   path: '/home/node/app/package-lock.json'
2023-04-21T18:50:06.751482141Z npm ERR! }
2023-04-21T18:50:06.751524014Z npm ERR! 
2023-04-21T18:50:06.751547062Z npm ERR! The operation was rejected by your operating system.
2023-04-21T18:50:06.751559056Z npm ERR! It is likely you do not have the permissions to access this file as the current user
2023-04-21T18:50:06.751573219Z npm ERR! 
2023-04-21T18:50:06.751578482Z npm ERR! If you believe this might be a permissions issue, please double-check the
2023-04-21T18:50:06.751589524Z npm ERR! permissions of the file and its containing directories, or try running
2023-04-21T18:50:06.751607345Z npm ERR! the command again as root/Administrator.
2023-04-21T18:50:06.752914701Z 
2023-04-21T18:50:06.752978777Z npm ERR! A complete log of this run can be found in: /home/node/.npm/_logs/2023-04-21T18_50_06_645Z-debug-0.log
```

However, it turns out this was just because the volume path needed to be updated in `docker-compose.fake-cht.yml`.  Can you try again with a completely fresh checkout of https://github.com/medic/cht-monitoring/tree/20_fix_internal_links and see if you still face the same issue?

hrm - that is not a fix for me - I still get the error.  Here's exact steps I took (from bash history):

```
git clone git@github.com:medic/cht-monitoring.git deleteme
cd deleteme
gco 20_fix_internal_links
cp cht-instances.example.yml cht-instances.yml\ncp grafana/grafana.example.ini grafana/grafana.ini\nmkdir -p grafana/data && mkdir  -p prometheus/data \nsudo apt install -y wamerican  # ensures /usr/share/dict/words is present for shuf call below \ncp .env.example .env\n
vim cht-instances.yml
docker compose -f docker-compose.yml -f development/fake-cht/docker-compose.fake-cht.yml up -d
docker kill $(docker ps -q)
docker logs deleteme-fake-cht-1
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /home/node/app/node_modules
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/home/node/app/node_modules'
npm ERR!  [Error: EACCES: permission denied, mkdir '/home/node/app/node_modules'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/home/node/app/node_modules'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.
```

--- a/development/README.md
+++ b/development/README.md
@@ -30,7 +30,7 @@ Set `http://fake-cht:8081` in the [list in `cht-instances.yml`](../cht-instances
 From the root directory, run:
 
 ```
-docker compose -f docker-compose.yml -f tests/fake-cht/docker-compose.fake-cht.yml up -d
+docker compose -f docker-compose.yml -f development/fake-cht/docker-compose.fake-cht.yml up -d
 ```
 
 ### Historical data
@@ -44,12 +44,12 @@ Each test is associated with an `xlsx` file in this directory that contains the 
 Start a fresh deployment of cht-monitoring without providing any CHT URL and with the test override:
 
 ```
-docker compose -f docker-compose.yml -f tests/docker-compose.test-data.yml up -d
+docker compose -f docker-compose.yml -f development/docker-compose.test-data.yml up -d
 ```
 
-Open the `xlsx` file of the test you want to run. Switch to the `data` sheet and Save As a `csv` file (named `data.csv`) in the `tests` directory.
+Open the `xlsx` file of the test you want to run. Switch to the `data` sheet and Save As a `csv` file (named `data.csv`) in the `development` directory.
 
-In your terminal, navigate to the `tests` directory and run `cat data.csv | ./generate_test_data.js > ../prometheus/data/data.txt`. This converts the `csv` file to the OpenMetrics format that Prometheus expects and injects it into the Prometheus data volume. _(This is a good point to stop and double-check the `data.txt` file to make sure it looks correct.)_
+In your terminal, navigate to the `development` directory and run `cat data.csv | ./generate_test_data.js > ../prometheus/data/data.txt`. This converts the `csv` file to the OpenMetrics format that Prometheus expects and injects it into the Prometheus data volume. _(This is a good point to stop and double-check the `data.txt` file to make sure it looks correct.)_
 
 Run `docker compose exec prometheus promtool tsdb create-blocks-from openmetrics /prometheus/data.txt /prometheus && docker compose restart prometheus` to push the data into Prometheus.
 
@@ -68,6 +68,6 @@ enabled = true
 host = maildev:1025
 ```
 
-Start the `maildev` server along with the rest of the monitoring stack by running `docker compose -f docker-compose.yml -f tests/docker-compose.smtp.yml up -d`.
+Start the `maildev` server along with the rest of the monitoring stack by running `docker compose -f docker-compose.yml -f development/docker-compose.smtp.yml up -d`.
 
 You can view the MailDev UI at http://localhost:1080.

--- a/development/fake-cht/docker-compose.fake-cht.yml
+++ b/development/fake-cht/docker-compose.fake-cht.yml
@@ -6,7 +6,7 @@ services:
     user: "node"
     working_dir: /home/node/app
     volumes:
-      - ./tests/fake-cht:/home/node/app
+      - ./development/fake-cht:/home/node/app
     networks:
       - cht-monitoring-net
     command: >

--- a/grafana/provisioning/dashboards/CHT/cht_admin_overview.json
+++ b/grafana/provisioning/dashboards/CHT/cht_admin_overview.json
@@ -257,7 +257,7 @@
           "links": [
             {
               "title": "DB Doc Count",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=2&var-cht_instance=${cht_instance}&var-db_name=${__field.labels.db}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=2&var-cht_instance=${cht_instance}&var-db_name=${__field.labels.db}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -416,7 +416,7 @@
           "links": [
             {
               "title": "Monthly Active Users",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=29&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=29&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -501,7 +501,7 @@
             {
               "targetBlank": false,
               "title": "Doc Conflicts",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=9&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=9&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -599,7 +599,7 @@
           "links": [
             {
               "title": "Fragmentation",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=19&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=19&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -694,7 +694,7 @@
           "links": [
             {
               "title": "Server Time",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=36&var-cht_instance=${cht_instance}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=36&var-cht_instance=${cht_instance}"
             }
           ],
           "mappings": [
@@ -800,7 +800,7 @@
           "links": [
             {
               "title": "Outbound Push Backlog",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=66&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=66&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -891,7 +891,7 @@
           "links": [
             {
               "title": "Sentinel Backlog",
-              "url": "./d/hkQUbyfVk/cht-admin-details?viewPanel=69&orgId=1&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?viewPanel=69&orgId=1&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -981,7 +981,7 @@
           "links": [
             {
               "title": "Users Over Replication Limit",
-              "url": "./d/hkQUbyfVk/cht-admin-details?viewPanel=68&orgId=1&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?viewPanel=68&orgId=1&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -1071,7 +1071,7 @@
           "links": [
             {
               "title": "Client Feedback/Errors",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=41&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=41&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [],
@@ -1405,7 +1405,7 @@
           "links": [
             {
               "title": "Failed Messages",
-              "url": "./d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=63&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
+              "url": "/d/hkQUbyfVk/cht-admin-details?orgId=1&viewPanel=63&var-cht_instance=${cht_instance}&from=${__from}&to=${__to}"
             }
           ],
           "mappings": [

